src/cycax/cycad/Part.py

```diff
@@ -0,0 +1,282 @@
+from Location import Location
+from Holes import Holes
+from Rectange_Cut_Out import Rectangle_Cut_Out
+from Nut_Cut_Out import Nut_Cut_Out
+from Slot import Slot
+
+LEFT = "LEFT"
+RIGHT = "RIGHT"
+TOP = "TOP"
+BOTTOM = "BOTTOM"
+FRONT = "FRONT"
+BACK = "BACK"
+
+
+class Part(Location):
+    def __init__(
+        self,
+        x: float,
+        y: float,
+        z: float,
+        side: str,
+        part_no: str,
+        x_size: float,
+        y_size: float,
+        z_size: float,
+    ):
+        super().__init__(x, y, z, side)
+        self.part_no = part_no
+        self.x_size = x_size
+        self.y_size = y_size
+        self.z_size = z_size
+        self.features = []  # Stores all the holes to be cut
+        self.move_holes = []
+        self.x_max: float = self.x_size  # Right
+        self.x_min: float = 0  # Left
+        self.y_min: float = 0  # Front
+        self.y_max: float = self.y_size  # Back
+        self.z_min: float = 0  # Bottom
+        self.z_max: float = self.z_size  # Top
+        self.bounding_box = {}
+        self.rotmax = [self.x_size, self.y_size, self.z_size]
+        self.moves = [0, 0, 0]
+        self.rotate = [0, 0, 0]
+        self.final_location = False
+
+    def make_hole(
+        self,
+        x: float,
+        y: float,
+        side: str,
+        diameter: float,
+        depth: float,
+        Inner: bool = True,
+    ):
+        """!!!!!THIS METHOD WILL ONLY WORK IF WE ARE MAKING HOLES IN THE CENTRE OF A CUBIC OBJECT, NEED TO RETHINK LOGIC!!!!!!
+        If instead of top and bottom it were possible to think rather (x, y, z_max) it would be more affective as the guessing wouldn't need to occur. These if statements wouldn't and is would work even when not a square.
+        It would however require more lines of code to be written on the user end."""
+        if side == TOP:
```

Comment:
A pattern here to reduce duplication could be to:
```
if side in [TOP, BOTTOM]:
   do
elif side in [LEFT, RIGHT]:
  do
```

Reply:
The top and bottom do not have the same methods. top=max_z, bottom=min_z. But I will keep it in mind for future.